docs/src/man/basics.md

```diff
@@ -1510,32 +1510,96 @@ julia> german[Not(5), r"S"]
-  columns;
-- `select!`: the same as `select` but updates the passed data frame in place;
-- `transform`: the same as `select` but keeps the columns that were already present in the data frame
-  (note though that these columns can be potentially modified by the transformation passed to `transform`);
-- `transform!`: the same as `transform` but updates the passed data frame in place.
-
-The fundamental ways to specify a transformation are:
-
-- `source_column => transformation => target_column_name`;
-  In this scenario the `source_column` is passed as an argument to `transformation` function
-  and stored in `target_column_name` column.
-- `source_column => transformation`;
-  In this scenario we apply the transformation function to `source_column` and the
-  target column names is automatically generated.
-- `source_column => target_column_name` renames the `source_column` to `target_column_name`.
-- `source_column` just keep the source column as is in the result without any transformation;
-
-These rules are typically called transformation mini-language.
-
+In DataFrames.jl there are seven functions that can be used to transform data frame columns:
+
+| Function     | Memory Usage                     | Column Retention                             | Row Retention                                       |
+| ------------ | -------------------------------- | -------------------------------------------- | --------------------------------------------------- |
+| `transform`  | Creates a new data frame.        | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
+| `transform!` | Modifies an existing data frame. | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
+| `select`     | Creates a new data frame.        | Retains only transformed columns.            | Retains same number of rows as source data frame.   |
+| `select!`    | Modifies an existing data frame. | Retains only transformed columns.            | Retains same number of rows as source data frame.   |
+| `subset`     | Creates a new data frame.        | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `subset!`    | Modifies an existing data frame. | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `combine`    | Creates a new data frame.        | Retains only transformed columns.            | Number of rows is determined by the transformation. |
+
+### Constructing Transformation Pairs
+All of the functions above use the same syntax which is commonly
+`transformation_function(source_dataframe, transformation)`.
+The `transformation` argument is a `Pair` which defines the
+transformation to be applied to the `source_dataframe`,
+and it can take any of the forms listed below:
+
+- `source_column_selector => function`
```

Comment:
maybe show the cases here in the order you explain them later (and hint that you are going to add these explanations)

Also add that these are the most common patterns.

Reply:
Also, better say `source_columns_selector` or just `source_columns` as multiple columns can be selected.

Reply:
I will keep `selector` as this was one of the main confusions I had. You do not use the source columns themselves (e.g. `df.x`). To me `column` just indicates the type of `selector`, and a `selector` could select one or more columns. `columns_selector` feels a little clunky to me personally.

Reply:
Maybe `source_column_names` is better, but I don't know if `Not()` or `1` would count as names.

Reply:
If we agree on what I have proposed above I think `source_column_selector` is OK, and I would give several examples of it to the user (like `1`, `:a`, `"a"`, `[:a, :b]`, `Not("a")` etc.) in the place where you would define it so that it is clear what we mean.